braindecode/augmentation/base.py

```diff
@@ -0,0 +1,223 @@
+# Authors: Cédric Rommel <cpe.rommel@gmail.com>
+#
+# License: BSD (3-clause)
+
+from typing import List, Tuple, Any
+from numbers import Real
+
+import pandas as pd
+from sklearn.utils import check_random_state
+import torch
+from torch import Tensor, nn
+from torch.utils.data import DataLoader
+from torch.utils.data._utils.collate import default_collate
+
+from braindecode.augmentation._functionals import identity
+
+Batch = List[Tuple[torch.Tensor, int, Any]]
+Output = Tuple[torch.Tensor, torch.Tensor]
+
+
+class Transform(torch.nn.Module):
+    """ Basic transform class used for implementing data augmentation
+    operations
+
+    Parameters
+    ----------
+    operation : callable
+        A function taking arrays X, y (sample features and
+        target resp.) and other required arguments, and returning the
+        transformed X and y.
+    probability : float, optional
+        Float between 0 and 1 defining the uniform probability of
+        applying the operation. Set to 1.0 by default (e.g always apply the
+        operation).
+    magnitude : float | None, optional
+        Defines the strength of the transformation applied between 0 and 1 and
+        depends on the nature of the transformation and on its range. Some
+        transformations don't have any magnitude (=None). It can be equivalent
+        to another argument of object with more meaning. In case both are
+        passed, magnitude will override the latter. Defaults to None.
+    random_state: int, optional
+        Seed to be used to instatiate numpy random number generator instance.
+        Used to decide whether or not to transform given the probability
+        argument. Defaults to None.
+    *args: tuple
+        Arguments to be passed to operation.
+    **kwargs: dict, optional
+        Keyword arguments to be passed to operation.
+    """
+
+    def __init__(self, operation, probability=1.0, magnitude=None,
+                 random_state=None, *args, **kwargs):
+        super().__init__()
+        if self.forward.__func__ is Transform.forward:
+            assert callable(operation), "operation should be a ``callable``."
+
+        self.operation = operation
+        assert isinstance(probability, Real), (
+            f"probability should be a ``real``. Got {type(probability)}.")
+        assert probability <= 1. and probability >= 0., \
+            "probability should be between 0 and 1."
+        self._probability = probability
+        self.rng = check_random_state(random_state)
+        assert (
+            (
+                isinstance(magnitude, Real) and
+                0 <= magnitude <= 1
+            ) or magnitude is None
+        ), "magnitude can be either a float between 0 and 1 or None"
```

Comment:
I'm not sure I get it, sorry. You can test AssertionError on pytest too. Or is it because the condition tested is confusing and I should decompose it?